tasks/upload_processor.py

```diff
@@ -206,23 +199,27 @@ def process_impl_within_lock(
         pr = None
         report_service = ReportService(UserYaml(commit_yaml))
 
+        in_parallel = (
+            parallel_processing.is_experiment_parallel
+            or parallel_processing.is_fully_parallel
+        )
```

Comment:
This is combination of checks is done in 3 places. Maybe we can add a method/property to the `ParallelProcessing` dataclass.